src/coreclr-debug/debugConfigurationProvider.ts

```diff
@@ -59,6 +61,46 @@ export class DotnetDebugConfigurationProvider implements vscode.DebugConfigurati
                 return undefined;
             }
         }
+        
+        // We want to ask the user if we should run dotnet  dev-certs https --trust, but this doesn't work in a few cases --
+        // Linux -- not supported by the .NET CLI as there isn't a single root cert store
+        // VS Code remoting/Web UI -- the trusted cert work would need to happen on the client machine, but we don't have a way to run code there currently
+        // pipeTransport -- the dev cert on the server will be different from the client
+        if (!this.platformInformation.isLinux() && !vscode.env.remoteName && vscode.env.uiKind != vscode.UIKind.Web && !debugConfiguration.pipeTransport)
+        {
+            if(debugConfiguration.checkForDevCert === undefined && debugConfiguration.serverReadyAction)
+            {
+                debugConfiguration.checkForDevCert = true;
+            }
+
+            if (debugConfiguration.checkForDevCert)
+            {
+                hasDotnetDevCertsHttps(this.options.dotNetCliPaths).then(async (hasDevCert) => {
+                    if(!hasDevCert)
+                    {
+                        const result = await vscode.window.showInformationMessage(
+                            "The selected launch configuration is configured to launch a web browser but no trusted development certificate was found. Create a trusted self-signed certificate?", 
+                            'Yes', "Don't Ask Again", 'Not Now'
```

Comment:
>'Not Now' [](http://example.com/codeflow?start=54&length=9)

If I am reading the API docs correctly, I think you want to make at least this one a [`MessageItem`](https://github.com/microsoft/vscode/blob/c83eff40dd2c2115f5fd5e67a6ce304fbc6fdefa/src/vscode-dts/vscode.d.ts#L1939) so you can make the escape key be the equivalent to this.